exercise/submission_models.py

```diff
@@ -28,18 +29,29 @@ def get_queryset(self):
             .prefetch_related('submitters')
 
     def create_from_post(self, exercise, submitters, request):
+
+        submission_data_list = [
+                (key, value) for (key, value) in query_dict_to_list_of_tuples(request.POST)
+                if key != '__aplus__'
+        ]
+        try:
+            meta_data_dict = json.loads(request.POST.get('__aplus__','{}'))
+        except json.JSONDecodeError:
+            raise ValueError("invalid json for __aplus__")
+
         new_submission = Submission.objects.create(
```

Comment:
Now I wonder why no exception is caught from `Submission.objects.create`. I guess somebody wanted to particularly log errors if the submission is created, but saving the submitted files fails after it.

Reply:
Hmm. So the old interface actually was value, None or exception as `.create` can raise IntegrityError, which is a subclass of DatabaseError. Thus, we are actually fixing a bug here too :)